src/api/route-services/__mocks__/samples.js

```diff
@@ -70,11 +70,30 @@ const mockDeleteSamples = jest.fn((projectUuid, sampleUuid) => new Promise((reso
   resolve(OK());
 }));
 
+const mockAddSample = jest.fn(
+  (projectUuid, experimentId, body) => new Promise((resolve, reject) => {
+    if (!projectUuid
+      || !body
+      || !experimentId
+      || !body.projectUuid
+      || !body.uuid
+    ) {
+      const err = new Error('Invalid body');
```

Comment:
This invalid is being received from inside the route-service, is this fine or should it be handled by the api validation?

Reply:
I think it would be better if it can be handled by the API validation. It can be detected earlier and so we don't have to code a handler inside the API to take care of it. I added one yesterday for `POST /runGEm2s`. It requires `paramsHash` to be available in the body.

Reply:
Moving it there would also allow us to have a better mock for `mockAddSample`, I think.

Reply:
Yes, I'll add a new ticket for doing this for all the mocks (all of them have this pattern) and also for fixing the api validation